From ae6963ae61cb78e20033517a6d745f02775d26ab Mon Sep 17 00:00:00 2001
From: Chunguang Xu <brookxu@tencent.com>
Date: Thu, 25 Mar 2021 06:57:56 +0000
Subject: [PATCH 88/91] bfq: disable merging between different groups under
 better_fairness

In order to better guarantee the Qos for each group, we do not
allow queues of different groups to be merged.

Signed-off-by: Chunguang Xu <brookxu@tencent.com>
---
 block/bfq-iosched.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/block/bfq-iosched.c b/block/bfq-iosched.c
index e0e51a7fa..bcc6cc209 100644
--- a/block/bfq-iosched.c
+++ b/block/bfq-iosched.c
@@ -2669,6 +2669,9 @@ static bool bfq_may_be_close_cooperator(struct bfq_queue *bfqq,
 	if (!bfq_bfqq_sync(bfqq) || !bfq_bfqq_sync(new_bfqq))
 		return false;
 
+	if (!bfq_bfqq_may_inject(bfqq, new_bfqq))
+		return false;
+
 	return true;
 }
 
-- 
2.31.1.272.g89b43f80a5

